api/src/main/java/io/opentelemetry/trace/attributes/IntOrStringAttribute.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.attributes;
+
+import io.opentelemetry.trace.Span;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Defines the behavior for a span attribute with long values which are normally handled as integers
+ * or strings in most applications.
+ */
+@Immutable
+public class IntOrStringAttribute extends AbstractAttribute<Integer> {
```

Comment:
I don't like this class. Also, from the name I would have expected it not to parse the string but leave it as-is.

Reply:
This was copied from OpenTracing based on the assumption that it was added there because of need

Reply:
I challenge that assumption ;) Or rather, I have an alternative idea, see https://github.com/open-telemetry/opentelemetry-java/pull/758#discussion_r389513439

Reply:
Changed